test/Microsoft.ML.Tests/Transformers/TextFeaturizerTests.cs

```diff
@@ -88,5 +88,108 @@ public void TextTokenizationWorkout()
             CheckEquality("Text", "tokenized.tsv");
             Done();
         }
+
+
+        [Fact]
+        public void TextNormalizationAndStopwordRemoverWorkout()
+        {
+            string sentimentDataPath = GetDataPath("wikipedia-detox-250-line-data.tsv");
+            var data = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadText(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var invalidData = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadFloat(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var est = new TextNormalizer(Env,"text")
+                .Append(new WordTokenizer(Env, "text", "words"))
+                .Append(new StopwordRemover(Env, "words", "words_without_stopwords"));
+            TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
+
+            var outputPath = GetOutputPath("Text", "words_without_stopwords.tsv");
+            using (var ch = Env.Start("save"))
+            {
+                var saver = new TextSaver(Env, new TextSaver.Arguments { Silent = true });
+                IDataView savedData = TakeFilter.Create(Env, est.Fit(data.AsDynamic).Transform(data.AsDynamic), 4);
+                savedData = new ChooseColumnsTransform(Env, savedData, "text", "words_without_stopwords");
+
+                using (var fs = File.Create(outputPath))
+                    DataSaverUtils.SaveDataView(ch, saver, savedData, fs, keepHidden: true);
+            }
+
+            CheckEquality("Text", "words_without_stopwords.tsv");
+            Done();
+        }
+
+        [Fact]
+        public void WordBagWorkout()
+        {
+            string sentimentDataPath = GetDataPath("wikipedia-detox-250-line-data.tsv");
+            var data = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadText(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var invalidData = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadFloat(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var est = new WordBagEstimator(Env, "text", "bag_of_words").
+                Append(new WordHashBagEstimator(Env, "text", "bag_of_wordshash"));
+            //TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
```

Comment:
>//TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic); [](start = 12, length = 78)

This method is failing when the size of output vector is determined by the incoming IDataView. The method calls the fit method to learn the output schema where `EmptyDataView` is passed. When fitting on `EmptyDataView`, the size is zero which is making the output vector a variable sized vector...:)

Need to find a way to handle this case.

Reply:
Talked to @Zruty0 offline. Need to fix how the ngram output vector size is computed i.e. need to ensure that

---
In reply to: [218967284](https://github.com/dotnet/machinelearning/pull/953#discussion_r218967284) [](ancestors = 218967284)

Reply:
Why not do this in this PR?

---
In reply to: [218989146](https://github.com/dotnet/machinelearning/pull/953#discussion_r218989146) [](ancestors = 218989146,218967284)

Reply:
Actually, NgramTransform is computing metadata based on the size of ngrams. I need to deeply look into this to enable it for EmptyDataView. I am thinking to open an issue against this. What do you think?

---
In reply to: [219285825](https://github.com/dotnet/machinelearning/pull/953#discussion_r219285825) [](ancestors = 219285825,218989146,218967284)

Reply:
I have created #969 for tracking this.

---
In reply to: [219287182](https://github.com/dotnet/machinelearning/pull/953#discussion_r219287182) [](ancestors = 219287182,219285825,218989146,218967284)

Reply:
Is this an issue on empty inputs? Is it possible to test at least the rest?

---
In reply to: [219315899](https://github.com/dotnet/machinelearning/pull/953#discussion_r219315899) [](ancestors = 219315899,219287182,219285825,218989146,218967284)